airflow/datasets/__init__.py

```diff
@@ -185,6 +186,14 @@ def evaluate(self, statuses: dict[str, bool]) -> bool:
     def iter_datasets(self) -> Iterator[tuple[str, Dataset]]:
         raise NotImplementedError
 
+    def iter_dag_deps(self, *, source: str, target: str) -> Iterator[DagDependency]:
```

Comment:
Any concerns just naming this `iter_dag_dependencies` to match the return value type name?

Reply:
Nope, `iter_dag_dependencies` is good. Will update it